sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -129,7 +131,11 @@ public virtual IEnumerable<Response<KeyBase>> GetKeyVersions(string name, Cancel
 
         public virtual IAsyncEnumerable<Response<KeyBase>> GetKeyVersionsAsync(string name, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            if (string.IsNullOrEmpty(name)) throw new ArgumentNullException(nameof(name));
```

Comment:
nit:  You may wish to consider `ArgumentException` here as the value may empty and not, in fact, `null`.

Reply:
Makes sense, I will update